### PR TITLE
Add support for more rtnl attributes

### PR DESCRIPTION
**Pull request by tklauser**
Add support for the following rtnl attributes:

* IFLA_CARRIER_CHANGES
* IFLA_PHYS_PORT_NAME
* IFLA_PHYS_SWITCH_ID
* IFLA_GSO_MAX_SEGS

### CODE REVIEW

File: libnl-route-3.sym [hunk 1]
```diff
@@ -286,6 +286,7 @@ global:
 	rtnl_link_get_broadcast;
 	rtnl_link_get_by_name;
 	rtnl_link_get_carrier;
+	rtnl_link_get_carrier_changes;
```
Comment (thom311) [inline libnl-route-3.sym]: a section in the linker-version script must never be modified once a new API is released. That means, new symbols must always be added to the section for the next upcoming release, that is `libnl_3_2_29` at the moment. I mean, new symbols must be added at last section in this file.

File: lib/route/link.c [hunk 2]
```diff
@@ -2273,6 +2284,17 @@ uint8_t rtnl_link_get_carrier(struct rtnl_link *link)
 }
 
 /**
+ * Return carrier on/off changes of link object
+ * @arg link		Link object
+ *
+ * @return Carrier changes.
+ */
+uint32_t rtnl_link_get_carrier_changes(struct rtnl_link *link)
```
Comment (thom311) [inline lib/route/link.c]: libnl is not very consistent about this, but I think that in most cases getters should return an `int` value indicating success (negative values meaning error, `NLE_*`). Something like https://github.com/thom311/libnl/blob/993a62d4501a9b40461d9f1276a63246268b4a3a/lib/route/link/ppp.c#L197

Not sure about getters that return pointers like `rtnl_link_get_phys_switch_id`. But for getters that return integers, I think we shouldn't mix an error result with the value.

File: libnl-route-3.sym (hunk 1 above)
Comment (tklauser) [inline libnl-route-3.sym]: Ah yes, didn't realize that. Will change accordingly (also the other .sym entries)

File: lib/route/link.c (hunk 2 above)
Comment (tklauser) [inline lib/route/link.c]: Most of the other `IFLA_*` "base" link attributes use these simple getters AFAICS, so I thought I stick to the pattern to make usage consistent.

But if you prefer the `rtnl_link_ppp_get_fd` style, I'll update the integer getters accordingly.

Comment (thom311) [inline lib/route/link.c]: It's not clear what's best, and consistency should be aimed for... but if the return value is uint32_t, there is no space to encode an error. Or even to encode whether the attribute is present -- which might be relevant information. I think a `int` error code is better.

**Comment by thom311** [comment]
Merged: https://github.com/thom311/libnl/commit/44880b68d847894cc71ec0bc5f774c2ccd88a0bf

Thank you!

**Pull request closed without merging**